bip32/utils.py

```diff
@@ -125,10 +125,22 @@ def _derive_public_child(pubkey, chaincode, index):
     return child_pub.format(), payload[32:]
 
 
+def _ripemd160(data):
+    try:
+        rip = hashlib.new("ripemd160")
+        rip.update(data)
+        return rip.digest()
+    except BaseException:
```

Comment:
Is it to catch exception even on non-CPython? Because it looks like CPython would [always raise a `ValueError`](https://github.com/python/cpython/blob/b9634ac776c24bc4d4a57859d884a94cdfe16043/Lib/hashlib.py#L123) if it couldn't find the requested hash function.

(Note: i'm not asking to change it, was just curious about the usage of `BaseException`)

Reply:
I just saw it elsewhere (I think electrum), and I didn't bother making sure that it's _always_ a `ValueError` (anyway, if _any_ error happened, it doesn't feel wrong to use the custom implementation).